### ISSUE TITLE
googlecloudexporter: retries for traces

### ISSUE THREAD

**Issue by monikeu** [open-telemetry/opentelemetry-collector-contrib#29607]
### Describe the issue you're reporting

TL:DR: I'm looking for retry configuration for traces, but it seems that there isn't anything which allows to configure it - retry_on_failure was removed recently and available configuration as 'max_backoff' applies only to metrics export. 

Details:
Recently with 5ab2db614bd8d67219a4e79fd65422a38876399a commit retry_on_failure was removed, the message for changelog was:
`googlecloudexporter`: remove retry_on_failure from the googlecloud exporter. The exporter itself handles retries, and retrying can cause issues. (#57233)

Looking at the  [doc ](https://github.com/GoogleCloudPlatform/opentelemetry-operations-go/blob/c9157213897e63eec1bf71d59a39c544530d79da/exporter/collector/README.md#L89) it seems that it was removed due to Google Cloud Monitoring related problems

Q1: Does it mean exporter -> Google Cloud Trace connection was not having similar problems as Monitoring?

I was looking for the trace export configs in opentelemetry-operations-go but i found only 'max_backoff' config param which applies only to metrics export. 

Q2: Is there any config for trace retries available?

Q3: Are there retries for traces available at all?

Please forgive me if i missed some part of docs/configs ;)

**Comment by monikeu** [comment]
_I believe we could provide such configuration to override the defaults_ -> that would be great.

We were facing increased 'otelcol_exporter_send_failed_spans'  metric value while having logs:

"failed to export to Google Cloud Trace: context deadline exceeded" - that is why we wanted to check out the timeouts of exporter. 

"otelcol_exporter_send_failed_spans" is described as

"Sustained rates of `otelcol_exporter_send_failed_spans` and
`otelcol_exporter_send_failed_metric_points` indicate that the Collector is not
able to export data as expected.
It doesn't imply data loss per se since there could be retries but a high rate
of failures could indicate issues with the network or backend receiving the
data." 

and we wanted to ensure that we do not face trace loss (or minimal) - do you know maybe if the google cloud exporter populates these metrics in case of permanent sent failure (when all retries failed) or on each retry attempt fail?

**Comment by nabheet** [comment]
> Tuning retry wouldn't help in this case, as it isn't hitting the retry limit. It is hitting the 12s default timeout, which you can configure by setting `timeout`.

I can't seem to find an example showing usage of this `timeout` field on `googlecloud` exporter. Do you have an example?

**Comment by dashpole** [comment]
You can see this from our unit tests: https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/f9b8ff45c4cbe3e36c0258e1ad0984d2ce1a2d7e/exporter/googlecloudexporter/testdata/config.yaml#L5
